frontend/packages/kubevirt-plugin/src/constants/select-dropdown-object-enum.ts

```diff
@@ -0,0 +1,28 @@
+import { ObjectEnum } from './object-enum';
+
+export type SelectDropdownData = {
+  description?: string;
+  label?: string;
+  order?: number;
+};
+
+export abstract class SelectDropdownObjectEnum<T> extends ObjectEnum<T> {
+  private readonly label: string;
+
+  private readonly description: string;
+
+  private readonly order: number;
+
+  protected constructor(value: T, { description, label, order }: SelectDropdownData = {}) {
+    super(value);
+    this.label = label;
+    this.description = description;
+    this.order = order;
+  }
+
+  getLabel = () => this.label;
```

Comment:
can we use toString method here instead to be consistent. We use this approach in other places as well.